### PR TITLE
zeek: rename from bro, update to 3.1.4

### PR DESCRIPTION
**Pull request by abenson**

### CODE REVIEW

**Comment by abenson** [comment]
I asked in #xbps whether to use replaces= or what, and was told transitional package instead

**Comment by sgn** [comment]
> replaces='bro>=0"?

`xbps-remove -o` will remove zeek if using replaces,
because it's not explicitly installed and no packages depends on it.

If xbps-pkgdb can mark zeek as manual when bro is manual, I think it's fine.

**Comment by abenson** [comment]
Yeah, replaces really should should make the package inherit installation state.

File: srcpkgs/zeek/template [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ musl-legacy-compat"
+short_desc="Advanced framework for network traffic analysis"
+maintainer="Andrew Benson <abenson+void@gmail.com>"
+license="BSD-3-Clause"
+homepage="https://www.zeek.org"
+distfiles="https://old.zeek.org/downloads/zeek-${version}.tar.gz"
+checksum=a2ef5f36dc4566d2ba129f34c14c269619b9797725b65d2696c27074db5f3e6a
+nocross="Needs to build part of itself to run, can't be built separately yet."
+
+case "$XBPS_TARGET_MACHINE" in
```
Comment (pullmoll) [inline srcpkgs/zeek/template:18]:
```
if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
    makedepends+=" musl-fts-devel"
    configure_args+=" -DIS_VOID_MUSL=1"
fi
```

File: srcpkgs/bro/template [hunk 2]
```diff
@@ -1,49 +1,11 @@
...
-hostmakedepends="flex pkg-config python3"
-makedepends="bind-devel geoip-devel libpcap-devel libressl-devel jemalloc-devel"
-short_desc="Advanced framework for network traffic analysis"
+revision=2
+archs=noarch
+build_style=meta
+depends="zeek"
+short_desc="Advanced framework for network traffic analysis (transitional dummy package)"
 maintainer="Andrew Benson <abenson+void@gmail.com>"
 license="BSD-3-Clause"
```
Comment (abenson) [inline srcpkgs/bro/template:10]: What would be the correct license to put here?

Comment (Piraty) [reply srcpkgs/bro/template:10]: [Manual.md](https://github.com/void-linux/void-packages/blob/7c361a68838e462bea726d95534de877c1426460/Manual.md) states: 
```
Empty meta-packages that don't include any files which thus have and require no license, should have set license="BSD-2-Clause".
```

**Comment by abenson** [comment]
Travis fails due to license. Otherwise seems g2g.

**Comment by Chocimier** [comment]
Please make bro subpackage of zeek.

**Comment by abenson** [comment]
Hm, this doesn't seem right.

    SONAME libbroker.so.2 from /usr/lib/libbroker.so.1.3

Though better than the same issue in the current bro build.

    /usr/lib/libbroker.so.0 -> /usr/lib/libbroker.so..

File: srcpkgs/zeek/patches/fix-musl.patch [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+-# Alpine support
+-if ( ${CMAKE_SYSTEM_NAME} MATCHES Linux AND EXISTS /etc/os-release )
+-    execute_process(
+-        COMMAND grep -q alpine /etc/os-release
+-        RESULT_VARIABLE os_release_alpine
+-    )
+-
+-    if ( os_release_alpine EQUAL 0 )
++# Void-musl support
++if ( IS_VOID_MUSL )
```
Comment (Piraty) [inline srcpkgs/zeek/patches/fix-musl.patch:16]: why not make this a generic var and even upstream that change?

Comment (sgn) [reply srcpkgs/zeek/patches/fix-musl.patch:16]: +1 Something like `NEED_LIBFTS` is way better then send that patch upstream.

File: srcpkgs/zeek/template [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	build_style=meta
+	archs=noarch
+	depends="${sourcepkg}>=${version}_${revision}"
+}
+
+binpac-devel_package() {
+	short_desc+=" - protocol parser compiler library (transitional)"
+	build_style=meta
+	archs=noarch
+	depends="${sourcepkg}>=${version}_${revision}"
```
Comment (sgn) [inline srcpkgs/zeek/template:56]: `binpac-devel` should `depends` on `zeek-devel`

File: srcpkgs/zeek/template [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# Template file for 'zeek'
+pkgname=zeek
+version=3.1.3
+revision=1
+archs="x86_64* i686* aarch64* armv7* ppc64*"
+build_style=cmake
+hostmakedepends="flex pkg-config python3"
+makedepends="bind-devel geoip-devel libpcap-devel libressl-devel jemalloc-devel"
```
Comment (sgn) [inline srcpkgs/zeek/template:8]: Doesn't `jemalloc-devel` only require if `-DENABLE_JEMALLOC=ON`?

**Comment by sgn** [comment]
zeek is easier to cross-compile than bro:
```diff
diff --git a/srcpkgs/zeek/template b/srcpkgs/zeek/template
index 7bee0e5bf2..52156006b0 100644
--- a/srcpkgs/zeek/template
+++ b/srcpkgs/zeek/template
@@ -4,21 +4,38 @@ version=3.1.3
 revision=1
 archs="x86_64* i686* aarch64* armv7* ppc64*"
 build_style=cmake
-hostmakedepends="flex pkg-config python3"
-makedepends="bind-devel geoip-devel libpcap-devel libressl-devel jemalloc-devel"
+hostmakedepends="flex pkg-config python3 libpcap-devel libressl-devel
+ bind-devel zlib-devel"
+makedepends="bind-devel geoip-devel libpcap-devel libressl-devel
+ jemalloc-devel zlib-devel"
 short_desc="Advanced framework for network traffic analysis"
 maintainer="Andrew Benson <abenson+void@gmail.com>"
 license="BSD-3-Clause"
 homepage="https://www.zeek.org"
 distfiles="https://old.zeek.org/downloads/zeek-${version}.tar.gz"
 checksum=d7bf24615c4c0af2435c99c9fb8c9c0f0ecdce375e184ba7f63b715ae5900a61
-nocross="Needs to build part of itself to run, can't be built separately yet."
 
 if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
 	makedepends+=" musl-fts-devel musl-legacy-compat"
 	configure_args+=" -DIS_VOID_MUSL=1"
 fi
 
+pre_configure() {
+	if [ "$CROSS_BUILD" ]; then
+		mkdir -p native
+		cd native
+		CC=$CC_FOR_BUILD CFLAGS="$CFLAGS_FOR_BUILD" \
+		CXX=$CXX_FOR_BUILD CXXFLAGS="$CXXFLAGS_FOR_BUILD" \
+		LD=$LD_FOR_BUILD LDFLAGS="$LDFLAGS_FOR_BUILD" \
+		cmake ..
+		make ${makejobs} binpac bifcl
+		configure_args+="
+		 -DBIFCL_EXE_PATH:PATH=${wrksrc}/native/aux/bifcl/bifcl
+		 -DBINPAC_EXE_PATH:PATH=${wrksrc}/native/aux/binpac/src/binpac"
+		cd ..
+	fi
+}
+
 post_install() {
 	vsv zeek
 	vlicense COPYING
```

**Pull request merged**